frontend/internal-packages/schema-bench/src/workspace/evaluation/evaluation.ts

```diff
@@ -1,71 +1,112 @@
+  EvaluationConfig,
+  EvaluationResult,
+  WorkspaceError,
+  WorkspaceResult,
+} from '../types'
+
+const loadOutputData = (
+  workspacePath: string,
+): WorkspaceResult<Map<string, Schema>> => {
   const outputDir = path.join(workspacePath, 'execution', 'output')
   const outputData = new Map<string, Schema>()
 
   if (!fs.existsSync(outputDir)) {
-    throw new Error(`Output directory does not exist: ${outputDir}`)
+    return err({ type: 'DIRECTORY_NOT_FOUND', path: outputDir })
   }
 
-  const files = fs
-    .readdirSync(outputDir)
-    .filter((file) => file.endsWith('.json'))
-
-  for (const file of files) {
-    const caseId = path.basename(file, '.json')
-    const filePath = path.join(outputDir, file)
-
-    try {
-      const content = fs.readFileSync(filePath, 'utf-8')
-      const schema = JSON.parse(content) as Schema
-      outputData.set(caseId, schema)
-    } catch (error) {
-      console.error(`❌ Failed to load output file ${file}:`, error)
-      throw error
+  try {
+    const files = fs
+      .readdirSync(outputDir)
+      .filter((file) => file.endsWith('.json'))
+
+    for (const file of files) {
+      const caseId = path.basename(file, '.json')
+      const filePath = path.join(outputDir, file)
+
+      try {
+        const content = fs.readFileSync(filePath, 'utf-8')
+        const schema = JSON.parse(content) as Schema
+        outputData.set(caseId, schema)
+      } catch (error) {
+        return err({
+          type: 'JSON_PARSE_ERROR',
+          path: filePath,
+          cause: error instanceof Error ? error.message : 'Unknown error',
+        })
+      }
     }
-  }
 
-  return outputData
+    return ok(outputData)
+  } catch (error) {
+    return err({
+      type: 'FILE_READ_ERROR',
+      path: outputDir,
+      cause: error instanceof Error ? error.message : 'Unknown error',
+    })
+  }
 }
 
-const loadReferenceData = (workspacePath: string): Map<string, Schema> => {
+const loadReferenceData = (
+  workspacePath: string,
+): WorkspaceResult<Map<string, Schema>> => {
   const referenceDir = path.join(workspacePath, 'execution', 'reference')
   const referenceData = new Map<string, Schema>()
 
   if (!fs.existsSync(referenceDir)) {
-    throw new Error(`Reference directory does not exist: ${referenceDir}`)
+    return err({ type: 'DIRECTORY_NOT_FOUND', path: referenceDir })
   }
 
-  const files = fs
-    .readdirSync(referenceDir)
-    .filter((file) => file.endsWith('.json'))
-
-  for (const file of files) {
-    const caseId = path.basename(file, '.json')
-    const filePath = path.join(referenceDir, file)
-
-    try {
-      const content = fs.readFileSync(filePath, 'utf-8')
-      const schema = JSON.parse(content) as Schema
-      referenceData.set(caseId, schema)
-    } catch (error) {
-      console.error(`❌ Failed to load reference file ${file}:`, error)
-      throw error
+  try {
+    const files = fs
+      .readdirSync(referenceDir)
+      .filter((file) => file.endsWith('.json'))
```

Comment:
FYI: In such cases, `Result.fromThrowable` can be used to reduce try-catch.

https://github.com/supermacro/neverthrow?tab=readme-ov-file#resultfromthrowable-static-class-method